Refactor documentation (#278)
nf-core/sarek: docs/usage.md

### -resume

Specify this when restarting a pipeline.
`Nextflow` will used cached results from any pipeline steps where the inputs are the same, continuing from where it got to previously.

You can also supply a run name to resume a specific run: `-resume [run-name]`. Use the `nextflow log` command to show previous run names.

> 

Author: Not sure how to phrase that.

> 

Reviewer: ```suggestion
You can also supply a run name or a session ID to resume a specific run: `-resume [run-name/session id]`. Use the `nextflow log` command to show previous run names and session IDs.
```

## Pipeline specific arguments

### --step

> **NB** only one step must be specified

Use this to specify the starting step:
Default `mapping`
Available: `mapping`, `prepare_recalibration`, `recalibrate`, `variant_calling`, `annotate`, `Control-FREEC`

> **NB** step can be specified with no concern for case, or the presence of `-` or `_`

### --input

Use this to specify the location of your input `TSV` (Tab Separated Values) file.

> (Note, the delimiter is the tab (`\t`) character, and no header are required)

Author: ```suggestion

> **NB** Delimiter is the tab (`\t`) character, and no header is required
```

> There are different kinds of `TSV` files that can be used as input, depending on the input files available (`FASTQ`, `unmapped BAM`, `reclibrated BAM`...).

Author: ```suggestion
There are different kinds of `TSV` files that can be used as input, depending on the input files available (`FASTQ`, `unmapped BAM`, `recalibrated BAM`...).
```

> `TSV` file should correspond to the correct step, see [`--step`](#--step) for more information.

Author: ```suggestion
The `TSV` file should correspond to the correct step, see [`--step`](#--step) for more information.
```

--input example_pair_fastq.tsv
```

#### Step: mapping from unmapped BAM

The `TSV` file for starting the mapping from `unmapped BAM` files should contain the columns:

- `subject sex status sample lane bam`

In this example (`example_ubam.tsv`), there are 3 read groups.

| | | | | | |
|-|-|-|-|-|-|
|SUBJECT_ID|XX|0|SAMPLE_ID|1|/samples/normal_1.bam|
|SUBJECT_ID|XX|0|SAMPLE_ID|2|/samples/normal_2.bam|
|SUBJECT_ID|XX|0|SAMPLE_ID|3|/samples/normal_3.bam|

```bash
--input example_ubam.tsv
```

Or, for a normal/tumor pair:

In this example (`example_pair_ubam.tsv`), there are 3 read groups for the normal sample and 2 for the tumor sample.

| | | | | | |
|-|-|-|-|-|-|
|SUBJECT_ID|XX|0|SAMPLE_ID1|1|/samples/normal_1.bam|
|SUBJECT_ID|XX|0|SAMPLE_ID1|2|/samples/normal_2.bam|
|SUBJECT_ID|XX|0|SAMPLE_ID1|3|/samples/normal_3.bam|
|SUBJECT_ID|XX|1|SAMPLE_ID2|1|/samples/tumor_1.bam|
|SUBJECT_ID|XX|1|SAMPLE_ID2|2|/samples/tumor_2.bam|

```bash
--input example_pair_ubam.tsv
```

#### Step: mapping from an input folder

Use this to specify the location to a directory with `FASTQ` files for the `mapping` step of a single germline sample only.
For example:

```bash
--input </path/to/directory>
```

> > **NB** All of the found `FASTQ` files are considered to belong to the sample.

Author: ```suggestion
> **NB** All of the found `FASTQ` files are considered to belong to the same sample.
```

### --nucleotides_per_second

Use this to estimate of how many seconds it will take to call variants on any interval, the default value is `1000` when not specified in the [`intervals`](#--intervals) file.

### --sentieon

[Sentieon](https://www.sentieon.com/) is a commercial solution to process genomics data with high computing efficiency, fast turnaround time, exceptional accuracy, and 100% consistency.

If [Sentieon](https://www.sentieon.com/) is available, use this `--sentieon` params to enable with `Sarek` to use some `Sentieon Analysis Pipelines & Tools`.
Adds the following tools for the [`--tools`](#--tools) options: `DNAseq`, `DNAscope` and `TNscope`.

Please refer to the [nf-core/configs](https://github.com/nf-core/configs#adding-a-new-pipeline-specific-config) repository on how to make a pipeline-specific configuration file based on the [munin-sarek specific configuration file](https://github.com/nf-core/configs/blob/master/conf/pipeline/sarek/munin.config).

Or ask us on the [nf-core Slack](http://nf-co.re/join/slack) on the following channels: [#sarek](https://nfcore.slack.com/channels/sarek) or [#configs](https://nfcore.slack.com/channels/configs).

> The following `Sentieon Analysis Pipelines & Tools` are available within `Sarek`.

Author: ```suggestion
The following `Sentieon Analysis Pipelines & Tools` are available within `Sarek`:
```

For example:

```bash
--tools Strelka,mutect2,SnpEff
```

Available variant callers: `ASCAT`, `Control-FREEC`, `FreeBayes`, `HaplotypeCaller`, `Manta`, `mpileup`, `MSIsensor`, `Mutect2`, `Strelka`, `TIDDIT`.

> **NB** tools can be specified with no concern for case

For more information on the individual variant callers, and where to find the variant calling results, check the [output](output.md) documentation.

> **WARNING** Not all variant callers are available for both germline and somatic variant calling.

For more information, please read the following documentation on [Germline variant calling](#germline-variant-calling), [Somatic variant calling with tumor - normal pairs](#somatic-variant-calling-with-tumor---normal-pairs) and [Somatic variant calling with tumor only samples](#somatic-variant-calling-with-tumor-only-samples)

Available annotation tools: `VEP`, `SnpEff`, `merge`.
For more details, please check the [annotation](#annotation-tools) documentation.

#### Germline variant calling

> Using Sarek, germline variant calling will always be performed if a variant calling tool with a germline mode is selected.

Author: ```suggestion
Using `Sarek`, germline variant calling will always be performed if a variant calling tool with a germline mode is selected.
```

Germline variant calling can currently only be performed with the following variant callers:

- *FreeBayes*
- *HaplotypeCaller*
- *Manta*
- *mpileup*
- *Sentieon*
- *Strelka*
- *TIDDIT*

For more information on the individual variant callers, and where to find the variant calling results, check the [output](output.md) documentation.

#### Somatic variant calling with tumor - normal pairs

> Using Sarek, somatic variant calling will be performed, if your input tsv file contains tumor / normal pairs (see [input](#--input) documentation for more information).

Author: ```suggestion
Using `Sarek`, somatic variant calling will be performed, if your input tsv file contains tumor / normal pairs (see [input](#--input) documentation for more information).
```

This may be useful if the qualities were very poor, or if there is some sort of unwanted bias at the 5' end.

### --clip_r2

Instructs `Trim Galore` to remove a number of bp from the 5' end of read 2 (paired-end reads only).
This may be useful if the qualities were very poor, or if there is some sort of unwanted bias at the 5' end.

### --three_prime_clip_r1

Instructs `Trim Galore` to remove a number of bp from the 3' end of read 1 (or single-end reads) AFTER adapter/quality trimming has been performed.
This may remove some unwanted bias from the 3' end that is not directly related to adapter sequence or basecall quality.

### --three_prime_clip_r2

Instructs `Trim Galore` to remove a number of bp from the 3' end of read 2 AFTER adapter/quality trimming has been performed.
This may remove some unwanted bias from the 3' end that is not directly related to adapter sequence or basecall quality.

### --trim_nextseq

This enables the option `--nextseq-trim=3'CUTOFF` within `Cutadapt`, which will set a quality cutoff (that is normally given with `-q` instead), but qualities of G bases are ignored.

> This trimming is in common for the `NextSeq` and `NovaSeq` platforms, where basecalls without any signal are called as high-quality G bases.

Reviewer: This trimming is ~~in~~ common for the `NextSeq` and `NovaSeq` platforms,

> This trimming is in common for the `NextSeq` and `NovaSeq` platforms, where basecalls without any signal are called as high-quality G bases.

Author: ```suggestion
This trimming is common for the `NextSeq` and `NovaSeq` platforms, where basecalls without any signal are called as high-quality G bases.
```

In order for the correct tagging to be performed, a read structure needs to be  specified as indicated below.

### --read_structure1

When processing `UMIs`, a read structure should always be provided for each of the `FASTQ` files, to allow the correct annotation of the `BAM` file.
If the read does not contain any `UMI`, the structure will be +T (i.e. only template of any length).
The read structure follows a format adopted by different tools, and described [here](https://github.com/fulcrumgenomics/fgbio/wiki/Read-Structures)

### --read_structure2

When processing `UMIs`, a read structure should always be provided for each of the `FASTQ` files, to allow the correct annotation of the `BAM` file.
If the read does not contain any UMI, the structure will be +T (i.e. only template of any length).
The read structure follows a format adopted by different tools, and described [here](https://github.com/fulcrumgenomics/fgbio/wiki/Read-Structures)

## Annotation

### --annotate_tools

Specify from which tools `Sarek` should look for `VCF` files to annotate, only for step `Annotate`.

> Available: `HaplotypeCaller`, `Manta`, `Mutect2`, `Strelka`, `TIDDIT`

Reviewer: add linebreak between default and available

### --save_reference

Enable saving reference indexes and other files built within `Sarek`.

```bash

> --save_reference

Reviewer: can be removed, I think. No subcommand provided here

### --species

This specifies the species used for running `VEP` annotation.

> If you use iGenomes or a local resource with `genomes.conf`, this has already been set for you appropriately.

Reviewer: maybe add example command similar to the other parameters
